src/Cryptol/TypeCheck/Solver/SMT.hs

```diff
@@ -100,7 +115,9 @@ startSolver onExit sCfg =
 
 -- | Shut down a solver instance
 stopSolver :: Solver -> IO ()
-stopSolver s = void $ SMT.stop (solver s)
+stopSolver s =
+  do void $ SMT.stop (solver s)
+     mapM_ hClose (loggerHandle s)
```

Comment:
If we make the change above, then we don't need to close the handle here.  I think keeping all the solver cleanup stuff in a single place is probably a good idea.